app/src/main/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragment.kt

```diff
@@ -20,41 +36,276 @@ class ScannerFragment : Fragment() {
+                        Log.d(TAG, "Waiting for binding to be initialized")
+                    };
+                    it.setSurfaceProvider(binding.scannerImagePreview.surfaceProvider)
+                }
+
+            // ImageCapture to save photo on device
+            imageCapture = ImageCapture.Builder().build()
+
+            // ImageAnalysis to detect text and objects
+            // TODO finish the image analyzer
+//            val imageAnalyzer = ImageAnalysis.Builder()
+//                .build()
+//                .also { it.setAnalyzer(cameraExecutor, setupImageAnalyzer()) }
+
+            // Select back camera as a default
+            val cameraSelector = CameraSelector.DEFAULT_BACK_CAMERA
+
+            try {
+                // Unbind use cases before rebinding
+                cameraProvider.unbindAll()
+
+                // Bind use cases to camera
+                cameraProvider.bindToLifecycle(
+                    this,
+                    cameraSelector,
+                    preview,
+                    imageCapture,
+                    // TODO finish the image analyzer
+//                    imageAnalyzer
+                )
+
+            } catch (exc: Exception) {
+                Log.e(TAG, "Use case binding failed while binding the camera to the view", exc)
+            }
+
+        }, ContextCompat.getMainExecutor(context))
+    }
+
+    // TODO finish the image analyzer
+    /**
+     * Setup the image analyzer to detect text and objects with proper callbacks
+     */
+//    private fun setupImageAnalyzer(): ImageAnalyzer {
+//        val textDetectedListener: TextDetectedListener = object : TextDetectedListener {
+//            override fun callback(text: Text) {
+//                scannerViewModel.detectTextSuccess(text)
+//            }
+//
+//            override fun errorCallback(exception: Exception) {
+//                scannerViewModel.detectTextError(exception)
+//            }
+//        }
+//
+//        val objectDetectedListener: ObjectDetectedListener = object : ObjectDetectedListener {
+//            override fun callback(text: String) {
+//                scannerViewModel.detectObjectSuccess(text)
+//            }
+//
+//            override fun errorCallback(exception: Exception) {
+//                scannerViewModel.detectObjectError(exception)
+//            }
+//        }
+//
+//        return ImageAnalyzer(textDetectedListener, objectDetectedListener)
+//    }
+
     override fun onDestroyView() {
         super.onDestroyView()
+        cameraExecutor.shutdown()
         _binding = null
     }
+
+    // Code adapted from https://developer.android.com/training/camerax
+    /**
+     * Companion object to hold constants
+     */
+    companion object {
+        // Tag for the [Log]
+        private const val TAG = "ScannerFragment"
+
+        // Format for the name of the file where the images will be saved
+        private const val FILENAME_FORMAT = "yyyy-MM-dd-HH-mm-ss-SSS"
```

Comment:
does the capitalization matter?

Reply:
of the format itslef (MM vs mm yes it matters), of the variable name FILENAME_FORMAT, no, it s just a way to define constant